src/sql/parts/notebook/models/notebookModel.ts

```diff
@@ -630,11 +630,11 @@ export class NotebookModel extends Disposable implements INotebookModel {
 			if (this.notebookOptions && this.notebookOptions.connectionService) {
 				let connectionService = this.notebookOptions.connectionService;
 				if (this._otherConnections) {
-					this._otherConnections.forEach(conn => connectionService.disconnect(conn).catch(e => console.log(e)));
+					notebookUtils.asyncForEach(this._otherConnections, async (conn) => await connectionService.disconnect(conn).catch(e => console.log(e)));
```

Comment:
Why this change? From reading the code I don't see much of a functional difference. Did you mean to await the call to `notebookUtils.asyncForEach` here?

Reply:
Yes. otherwise array.forEach wouldn't wait.